src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,1077 @@
+        self.__asert_message_json_from_status(substatus_file_data[0]["status"]["substatus"][0], patch_count, error_count=0)
+
+        # Assert no truncated status file
+        self.__assert_no_truncation_status_file(Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count)
+
+    def test_only_assessment_patches_truncation_over_size_limit(self):
+        """ Perform truncation on assessment patches.
+        Expecting:
+        assessment substatus status: warning,
+        assessment tombstone records,
+        assessment errors code: 2 (warning),
+        assessment errors details code: patches were truncated to limit reporting data volume. In-VM logs contain complete lists. """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = random.randint(780, 1000)
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        assessment_msg = json.loads(substatus_file_data[0]["status"]["substatus"][0]["formattedMessage"]["message"])
+        self.__assert_patch_summary_from_status(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, is_under_agent_size_limit=False)
+        self.__asert_message_json_from_status(substatus_file_data[0]["status"]["substatus"][0], patch_count, error_count=0)
+
+        # Assert assessment truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        assessment_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        # Assert length of truncated patches < length of complete status file patches
+        self.assertTrue(len(assessment_truncated_msg["patches"]) < patch_count)
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_WARNING)
+
+        # Assert truncated error
+        self.__assert_truncated_error(truncated_substatus_file_data[0]["status"]["substatus"][0], error_count=0)
+
+        # Assert all assessment fields in the message json are equal in both status files
+        self.__assert_assessment_truncated_msg_fields(assessment_msg, assessment_truncated_msg)
```

Comment:
None of these functions validate the file size change from before and after truncation.

And, from what I see __assert_patch_summary_from_status(), __assert_truncated_error(), __assert_assessment_truncated_msg_fields() can be combined into a single validation function which is used for all different scenarios.

Use our CRP BVTs, validation functions for reference.

Reply:
these are core assertion of all truncation tests
self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
self.assertTrue(truncated_substatus_file_byte_size < Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)

I don't see the benefit validate the file size change from before and after truncation because status file can be > 128kb and still less than complete status (create false positive test). the above assertions ensure the status file must be < 126kb and finally < 128kb (worst case, it will not happen unless there's code change), this means truncation applied and status file is < complete status file size.

anyway i've added assertion just to get this pr merge in. 
self.assertTrue(len(json.dumps(truncated_substatus_file_data)) < len(json.dumps(substatus_file_data)))

done

Reply:
The main goal of writing code/making changes should never be just to move a PR faster, It should be to implement the appropriate functionality.

Also, the other ask to combine __assert_patch_summary_from_status() and __assert_assessment_truncated_msg_fields() in the first message is still pending

Reply:
@feng-j678 is this still open?

Reply:
Pending conversation continued in a new comment